### PR TITLE
Multiple tree flow fix

### PR DESCRIPTION
**Pull request by ankitecd**
Changes in this pull request:
- Fixing multiple tree flow

### CODE REVIEW

**Comment by norbertschuler** [comment]
@ankitecd is this already ready to be tested?

**Comment by ankitecd** [comment]
> 
> 
> @ankitecd is this already ready to be tested?

No, not yet

**Comment by norbertschuler** [comment]
@ankitecd please add me as a reviewer once this is ready to be tested. 

- [ ] I already found some warning after having entered three points for a polygon and going to the page to register sample trees as following:

Screenshot_1615463113Screenshot_1615463200

- [ ] another error is I cannot yet upload my multiple tree registration to the backend with getting a 400 error as following: 
```
Error at: /action/upload, POST - /treemapper/plantLocations - > {
  "data": "Invalid type specified: 'multiple'. Must be one of: sample,single,multi",
  "status": 400,
  "config": {
    "url": "/treemapper/plantLocations",
    "method": "post",
    "data": "{\"type\":\"multiple\",\"captureMode\":\"on-site\",\"deviceLocation\":{\"coordinates\":[9.9355,53.5496],\"type\":\"Point\"},\"geometry\":{\"type\":\"Polygon\",\"coordinates\":[[[9.935356140136719,53.54993438720703],[9.935602188110352,53.54993438720703],[9.935342788696289,53.54953384399414],[9.93520450592041,53.54974365234375],[9.935356140136719,53.54993438720703]]]},\"plantDate\":\"2021-03-11\",\"registrationDate\":\"2021-03-11\",\"plantProject\":null,\"measurements\":{\"height\":null,\"width\":null},\"scientificSpecies\":[{\"treeCount\":20}]}",
    "headers": {
      "Accept": "application/json, text/plain, */*",
      "Content-Type": "application/json",
      "Authorization": "OAuth ***",
      "X-SESSION-ID": "da49dbcf-1b46-4c1b-a0c4-ab2218e9d6ea"
    },
    "baseURL": "https://app-development.plant-for-the-planet.org",
  },
}
```

- [ ] For registering sample trees I made photos, but they do not seem to get stored, do they?
- [ ] Also registered sample trees can be placed outside the given polygon (I think a test if they are placed inside would be nice, but maybe is a little more difficult to do.)

**Comment by norbertschuler** [comment]
- [ ] The GeoJson export just exports the polygon, no sample trees or other information, right?

Simulator Screen Shot - iPhone 11 - 2021-03-11 at 13 24 25

Simulator Screen Shot - iPhone 11 - 2021-03-11 at 13 22 02Simulator Screen Shot - iPhone 11 - 2021-03-11 at 13 22 07

**Comment by norbertschuler** [comment]
- [ ] Let's hide the button to import GeoJson (or add some hint that the function is not yet available):

Simulator Screen Shot - iPhone 11 - 2021-03-11 at 13 33 22

File: app/components/RegisterSingleTree/MapMarking.js [hunk 1]
```diff
@@ -107,11 +105,12 @@ const MapMarking = ({ updateScreenState, inventoryState, resetRouteStack }) => {
       recenterCoords = [location.coords.longitude, location.coords.latitude];
     }
     setIsInitial(true);
-    camera && camera.current.setCamera( {
-      centerCoordinate: recenterCoords,
-      zoomLevel: 18,
-      animationDuration: 1000,
-    });
+    camera &&
+      camera.current.setCamera({
```
Comment (norbertschuler) [inline app/components/RegisterSingleTree/MapMarking.js:109]: We should add check for all the other usages of the camera as well (I got a problem registering sample trees):

Bildschirmfoto 2021-03-12 um 10 37 13

**Review by norbertschuler — CHANGES_REQUESTED** [review]

**Comment by norbertschuler** [comment]
- [ ] iOS Pods have a problem to be installed:
```
[!] CocoaPods could not find compatible versions for pod "@react-native-mapbox-gl-mapbox-static":
  In snapshot (Podfile.lock):
    @react-native-mapbox-gl-mapbox-static (= 5.8.0, ~> 5.7)

  In Podfile:
    react-native-mapbox-gl (from `../node_modules/@react-native-mapbox-gl/maps`) was resolved to 8.1.0, which depends on
      @react-native-mapbox-gl-mapbox-static (~> 5.9.0)
```

File: package.json [hunk 2]
```diff
@@ -17,10 +17,12 @@
     "@react-native-community/geolocation": "^2.0.2",
     "@react-native-community/masked-view": "^0.1.7",
     "@react-native-community/toolbar-android": "0.1.0-rc.2",
-    "@react-native-mapbox-gl/maps": "^8.0.0",
+    "@react-native-mapbox-gl/maps": "^8.1.0",
```
Comment (norbertschuler) [inline package.json:20]: The iOS archive process fails using this version, see https://github.com/react-native-mapbox-gl/maps/issues/1097

Comment (norbertschuler) [reply package.json:20]: @ankitecd is there any special reason for using the new version 8.1.0 or can I downgrade to 8.0.0 again?

Comment (norbertschuler) [reply package.json]: I think I made the beta work.

Comment (ankitecd) [reply package.json]: > @ankitecd is there any special reason for using the new version 8.1.0 or can I downgrade to 8.0.0 again?

You can downgrade if required

Comment (norbertschuler) [reply package.json]: Beta is working.

**Comment by norbertschuler** [comment]
Repeating the conceptual issue here (and deleting them from the above comment) - @sagararyal can you please check and help to decide if this needs any changes?

- [ ] registered sample trees can have a species not selected for the registered trees
- [ ] registered sample trees can be placed outside the given polygon and the coordinates of the sample trees are not displayed any more
- [ ] registered sample trees are listed with a right arrow without any further functionality (e.g. display details?)

Simulator Screen Shot - iPhone 11 - 2021-03-22 at 09 20 16Simulator Screen Shot - iPhone 11 - 2021-03-22 at 09 25 33Simulator Screen Shot - iPhone 11 - 2021-03-22 at 09 20 25

**Comment by norbertschuler** [comment]
- [ ] There seems to be an error entering the height and diameter on the sample tree. I am sure I entered `2.22` and `2.1` as values:

Simulator Screen Shot - iPhone 11 - 2021-03-22 at 10 19 55

**Comment by norbertschuler** [comment]
I tested multiple tree registrations with tags and got the following issues:

- [ ] For the registered trees there is no possibility to enter a tag and so nothing is send with the API call to upload the registered trees and `tag: null` is returned - should it be possible to tag these trees?

- [ ] For the sample trees if you enter the tag in the dialog at the same time when entering the diameter and height the tag gets send with the API call to upload the sample tree, but if you enter the tag later on the review page, it is not send anymore with the API call to upload a sample tree:

Simulator Screen Shot - iPhone 11 - 2021-03-22 at 10 35 10Simulator Screen Shot - iPhone 11 - 2021-03-22 at 10 35 47

File: app/components/SampleTrees/SampleTreesReview.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+      <Image source={imageSource} style={styles.image} resizeMode={'stretch'} />
+      <View style={styles.specieListTextContainer}>
+        <View style={styles.specieHeadingContainer}>
+          <Text style={styles.specieListHeading}>{sampleTree.specieName}</Text>
+          <Text style={styles.specieListHeading}>
+            {sampleTree.tagId ? ' • ' + sampleTree.tagId : ''}
+          </Text>
+        </View>
+        <Text style={styles.subHeadingText}>
+          #{index + 1} • {sampleTree.specieHeight}cm • {sampleTree.specieDiameter}cm
```
Comment (norbertschuler) [inline app/components/SampleTrees/SampleTreesReview.js:28]: Please check if we need to have a distinction between cm and inches like done elsewhere depending on the users countryCode.

Comment (norbertschuler) [reply app/components/SampleTrees/SampleTreesReview.js:28]: Also height is measured in `m` (or `foot`) while diameter is measured in `cm` (or `inches`).

**Comment by ankitecd** [comment]
> 
> 
> Repeating the conceptual issue here (and deleting them from the above comment) - @sagararyal can you please check and help to decide if this needs any changes?
> 
>     * [ ]  registered sample trees can have a species not selected for the registered trees
> 
>     * [ ]  registered sample trees can be placed outside the given polygon and the coordinates of the sample trees are not displayed any more
> 
>     * [ ]  registered sample trees are listed with a right arrow without any further functionality (e.g. display details?)
> 
> 
> Simulator Screen Shot - iPhone 11 - 2021-03-22 at 09 20 16Simulator Screen Shot - iPhone 11 - 2021-03-22 at 09 25 33Simulator Screen Shot - iPhone 11 - 2021-03-22 at 09 20 25

1. yes sample tree can be registered even if the species for multiple trees is not registered. But for completing the multiple tree atleast one specie is required. That is by design.
2. Coordinates were not showing earlier also and by design it is not there.
3. This is fixed in new commit

**Comment by ankitecd** [comment]
> 
> 
> I tested multiple tree registrations with tags and got the following issues:
> 
>     * [ ]  For the registered trees there is no possibility to enter a tag and so nothing is send with the API call to upload the registered trees and `tag: null` is returned - should it be possible to tag these trees?
> 
>     * [ ]  For the sample trees if you enter the tag in the dialog at the same time when entering the diameter and height the tag gets send with the API call to upload the sample tree, but if you enter the tag later on the review page, it is not send anymore with the API call to upload a sample tree:
> 
> 
> Simulator Screen Shot - iPhone 11 - 2021-03-22 at 10 35 10Simulator Screen Shot - iPhone 11 - 2021-03-22 at 10 35 47

1. It is not possible to to tag multiple trees with one tag so tag for multiple tree is not required.
2. Fixed in latest commit.

**Comment by norbertschuler** [comment]
- [ ] I tested entering the data without being logged in in `cm`/`m` and then logged in with my account having countryCode=US and seeing the converted values - I miss some rounding with two places behind the dot (xxx.xx) for the `inches`/`feet` values:

Simulator Screen Shot - iPhone 11 - 2021-03-23 at 14 21 54Simulator Screen Shot - iPhone 11 - 2021-03-23 at 14 24 22Simulator Screen Shot - iPhone 11 - 2021-03-23 at 14 23 59

**Comment by norbertschuler** [comment]
> Simulator Screen Shot - iPhone 11 - 2021-03-23 at 14 21 54Simulator Screen Shot - iPhone 11 - 2021-03-23 at 14 24 22Simulator Screen Shot - iPhone 11 - 2021-03-23 at 14 23 59

- [ ] The review page for a sample tree is not showing the photo if it is available.

**Comment by norbertschuler** [comment]
- [ ] The picture is only showing the lower part of the image with the woman - is that correct?

Simulator Screen Shot - iPhone 11 - 2021-03-23 at 18 16 11Simulator Screen Shot - iPhone 11 - 2021-03-23 at 18 20 42

**Comment by norbertschuler** [comment]
- [ ] Species list has a green link for every number of species, but the link has not functionality (we should either remove the link or add the missing functionality):

Simulator Screen Shot - iPhone 11 - 2021-03-23 at 18 19 35

**Comment by norbertschuler** [comment]
- [ ] I am missing the inch/feet conversion in the sample tree detail screen:

Simulator Screen Shot - iPhone 11 - 2021-03-23 at 18 32 16!Simulator Screen Shot - iPhone 11 - 2021-03-23 at 18 32 21

**Comment by norbertschuler** [comment]
- [ ] If I enter values 1 for height and diameter in feet and inches, on the review screen they are shown differently. So either the input or the output is wrong here too:

Simulator Screen Shot - iPhone 11 - 2021-03-23 at 18 38 08Simulator Screen Shot - iPhone 11 - 2021-03-23 at 18 38 13

**Comment by norbertschuler** [comment]
> * [ ]  If I enter values 1 for height and diameter in feet and inches, on the review screen they are shown differently. So either the input or the output is wrong here too:
> 
> Simulator Screen Shot - iPhone 11 - 2021-03-23 at 18 38 08Simulator Screen Shot - iPhone 11 - 2021-03-23 at 18 38 13

- [ ] Now 1 inch / 1 feet got converted into this on the review screen - even with some calculations a value of `1` should not be converted into a value of `0.98`:

Simulator Screen Shot - iPhone 11 - 2021-03-24 at 13 46 00

**Comment by norbertschuler** [comment]
> * [x]  Species list has a green link for every number of species, but the link has not functionality (we should either remove the link or add the missing functionality):

- [ ] There is a similar small UI issue on the species selection screen as the species is written in green color, but is not a link to do anything clicking on it.

Simulator Screen Shot - iPhone 11 - 2021-03-24 at 13 42 10

**Comment by norbertschuler** [comment]
- [ ] If you do not change your location between point A and B, you get an alert, but it is shown on the next screen when you are ask to take picture and you can just click it away and ignore it - this is not really a bug, but also maybe not the wanted behavior @sagararyal:

Simulator Screen Shot - iPhone 11 - 2021-03-24 at 13 37 45Simulator Screen Shot - iPhone 11 - 2021-03-24 at 13 37 49

**Review by norbertschuler — APPROVED** [review]
All my issues seem to be solved, but I got 401 error while uploading the test data accumulated today and yesterday. Hitting the upload button twice seems to help, but I wonder how the OAuth token could get outdated between different uploads within a few seconds. Once I also got the error `Bugsnag could not notify: error must be of type Error` which might indicate that the parameter given to `bugsnag.notify()` somewhere in the code has the wrong type (also important to check for #262)?

I approved this PR now as I think these issues can be done in new PR not blocking this merge.

**Pull request merged**